File: app/src/main/java/ru/iandreyshev/parserrss/presentation/view/IFeedView.java
package ru.iandreyshev.parserrss.presentation.view;

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy;
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import ru.iandreyshev.parserrss.models.rss.ViewRss;

public interface IFeedView extends IBaseView {
    @StateStrategyType(AddToEndStrategy.class)
    void insertRss(final ViewRss rss);

    @StateStrategyType(AddToEndStrategy.class)
    void removeRss(final ViewRss rss);

    @StateStrategyType(SkipStrategy.class)
    void openArticle(long articleId);

    @StateStrategyType(SkipStrategy.class)
    void openAddingRssDialog();

    @StateStrategyType(AddToEndSingleStrategy.class)
    void startProgressBar(boolean isStart);

    @StateStrategyType(SkipStrategy.class)
    void openRssInfo(final ViewRss rss);

    @StateStrategyType(SkipStrategy.class)
    void onFeedUpdate();
}

File: app/src/main/java/ru/iandreyshev/parserrss/presentation/view/IFeedView.kt
package ru.iandreyshev.parserrss.presentation.view

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

import ru.iandreyshev.parserrss.models.rss.ViewRss

interface IFeedView : IBaseView {
    @StateStrategyType(AddToEndStrategy::class)
    fun insertRss(rss: ViewRss)

    @StateStrategyType(AddToEndStrategy::class)
    fun removeRss(rss: ViewRss)

    @StateStrategyType(SkipStrategy::class)
    fun openArticle(articleId: Long)

    @StateStrategyType(SkipStrategy::class)
    fun openAddingRssDialog()

    @StateStrategyType(AddToEndSingleStrategy::class)
    fun startProgressBar(isStart: Boolean)

    @StateStrategyType(SkipStrategy::class)
    fun openRssInfo(rss: ViewRss)

    @StateStrategyType(AddToEndSingleStrategy::class)
    fun openEmptyContentMessage(isOpen: Boolean)

    @StateStrategyType(SkipStrategy::class)
    fun openPage(position: Int)

    @StateStrategyType(AddToEndSingleStrategy::class)
    fun setToolbarScrollable(isScrollable: Boolean)
}
